src/test/java/com/basho/riak/client/core/operations/itest/ISearchTestBase.java

```diff
@@ -0,0 +1,195 @@
+package com.basho.riak.client.core.operations.itest;
+
+import com.basho.riak.client.api.RiakClient;
+import com.basho.riak.client.api.commands.buckets.StoreBucketProperties;
+import com.basho.riak.client.api.commands.kv.StoreValue;
+import com.basho.riak.client.api.commands.search.DeleteIndex;
+import com.basho.riak.client.api.commands.search.StoreIndex;
+import com.basho.riak.client.core.RiakFuture;
+import com.basho.riak.client.core.query.Location;
+import com.basho.riak.client.core.query.Namespace;
+import com.basho.riak.client.core.query.RiakObject;
+import com.basho.riak.client.core.query.search.YokozunaIndex;
+import com.basho.riak.client.core.util.BinaryValue;
+import org.junit.Assert;
+
+import java.util.concurrent.ExecutionException;
+
+import static junit.framework.Assert.assertTrue;
+
+/**
+ * Base class for testing Search features.
+ *
+ * @author Alex Moore  <amoore at basho dot com>
+ * @since 2.0.5
+ */
+public class ISearchTestBase extends ITestBase
+{
+    public static void setupSearchEnvironment(String bucketName, String indexName) throws ExecutionException, InterruptedException
+    {
+        final RiakClient client = new RiakClient(cluster);
+
+        if(legacyRiakSearch)
+        {
+            Namespace legacyNamespace = getLegacyNamespace(bucketName);
+            setupLegacySearchBucketProps(client, legacyNamespace);
+            setupData(client, legacyNamespace);
+        }
+
+        if(testYokozuna && testBucketType)
+        {
+            Namespace yokoNamespace = getYokoNamespace(bucketName);
+
+            setupYokoIndexAndBucketProps(client, yokoNamespace, indexName);
+            setupData(client, yokoNamespace);
+        }
+
+        // TODO: figure out a better way than Thread.sleep
+        // Give search some time to process everything
+        Thread.sleep(3000);
```

Comment:
Yup. Something better would be nice.  Can we interrogate riak somehow to see if it's ready?